Source/PlayerEx.cpp

```diff
@@ -29,7 +29,32 @@ struct OP2Player
 	int scientists;
 	int kids;
 	int boolRecalcValues;
-	int unk3[70];
+	int numAvailableWorkers;	//	56EFC0	0x0A4	numAvailableWorkers
+	int numAvailableScientists;
+	int amountPowerGenerated;
+	int inactivePowerCapacity;
+	int amountPowerConsumed;
+	int amountPowerAvailable;
+	int numIdleBuildings;
+	int numActiveBuildings;
+	int numBuildings;
+	int numUnpoweredStructures;
+	int numWorkersRequired;	// 56EFE8	0x0CC	numWorkersRequired[assigned workers and scientists]
+	int numScientistsRequired;	// 56EFEC	0x0D0	numScientistsRequired[assigned to buildings, or researching]
+	int numScientistsAsWorkers;
+	int numScientistsAssignedToResearch;
+	int totalFoodProduction;
+	int totalFoodConsumption;
+	int foodLacking;	// 56F000	0x0E4	foodLacking[how much food needed to stop people starving ? Note this can be 0 even if there is a food deficit]
+	int netFoodProduction;
+	int numSolarSatellites;	// 56F008	0x0EC	numSolarSatellites[copy from bitfield at offset 8]
+
+	// 56F00C	0x0F0	TotalCapacities
+	int totalRecreationFacilityCapacity;// 56F00C	 0x0F0	 totalRecreationFacilityCapacity
+	int totalForumCapacity;// 56F010	 0x0F4	 totalForumCapacity
+	int totalMedCenterCapacity;// 56F014	 0x0F8	 totalMedCenterCapacity
+	int totalResidenceCapacity;// 56F018	 0x0FC	 totalResidenceCapacity

```

Comment:
Should maybe use spaces to align the end of line comments. Looks like there is tab stop misalignment here.

Reply:
It's not indented at all. Pretty sure all of this information was from Ollydebug to help with figuring out where to put the variables. I can delete all of this unless anything looks useful.

Reply:
I would certainly prefer seeing the 3rd comment column removed as it just seems to repeat the variable name.

A couple of the comments expand on how the field works such as on numWorkersRequired. I wouldn't want to see those comments removed.

I don't know how clear it will be to someone the addresses came from ollydbg. If we think it is useful, maybe annotate it as such. I'm assuming the 2nd comment column is the memory address from the start of the structure. I don't really have an opinion one way or the other. 

I'm not a big fan of having to line everything up with spaces. It gets tedious and creates lots of commit noise when changing names if the spacing has to change significantly. It is visually pleasing. Maybe it is okay to just leave them the way they are if we want the comments? I'm not really opposed to seeing them spaced out for formatting either though.

Reply:
I agree about removing the third comment column that just restates the variable name.

I could go either way on the address and offset portion of the comments.